colossalai/auto_parallel/solver/graph_analysis.py

```diff
@@ -0,0 +1,210 @@
+            elif tensor_type == 'buffer':
+                iterator = module.named_buffers()
+            else:
+                raise ValueError(f"Expected tensor_type to be param or buffer, but got {tensor_type}")
+
+            for name, tensor in iterator:
+                tensor_name = f'{node.name}.{name}'
+
+                if not checked_variables.exists(tensor_name):
+                    live_tensor = LiveVariable(name=tensor_name, meta=tensor.to('meta'), is_inplace=False)
+                    unique_live_vars.append(live_tensor)
+                    checked_variables.append(live_tensor)
+                    all_live_variables.append(live_tensor)
+
+        for idx, node in enumerate(compute_nodes):
+            #############################
+            # find new living variables #
+            #############################
+            # detect whether the current op is an in-place op
+            # if it is an in-place op, we would deem it as a duplciate var
+            is_inplace = False
+            if node.op == 'call_function':
+                # check if this is an inplace op such as torch.nn.functional.relu(x, inplace=True)
+                if node.kwargs.get('inplace', False):
+                    is_inplace = True
+            elif node.op == 'call_module':
+                # to check if this is an inplace op such as torch.nn.Relu(inplace=True)
+                module = get_node_module(node)
+                if getattr(module, 'inplace', False):
+                    is_inplace = True
+
+            # add the output var
+            meta = getattr(node, '_meta_data', None)
+            live_var = LiveVariable(name=node.name, meta=meta, is_inplace=is_inplace)
+            if not is_inplace:
+                unique_live_vars.append(live_var)
+            checked_variables.append(live_var)
+            all_live_variables.append(live_var)
+
+            # add the model parameters
+            if node.op == 'call_module':
+                _add_param_or_buf(node, tensor_type='param')
+                _add_param_or_buf(node, tensor_type='buffer')
+
+            # add this output variable to the checked list
+            checked_variables.append(live_var)
+
+            # check if any input is not checked yet
+            for arg in node.args:
+                arg_name = str(arg)
+                if not checked_variables.exists(arg_name):
+                    meta = getattr(node, '_meta_data', None)
+                    live_var_from_arg = LiveVariable(name=arg_name, meta=meta, is_inplace=False)
+                    all_live_variables.append(live_var_from_arg)
+                    checked_variables.append(live_var_from_arg)
+                    unique_live_vars.append(live_var_from_arg)
+
+            # TODO: add the logic to remove live variables
+            # this should be completed if we are able to trace the backward compute graph
+
+            ###########################
+            # compute the memory cost #
```

Comment:
This memory cost without sharding strategy info won't be used in solver, and it is hard to enumerate the memory cost of all possible strategy combinations. So I think we could just leave the memory cost computing to solver as memory constraint.

Reply:
Sure.

Reply:
I have removed the code related to memory cost.